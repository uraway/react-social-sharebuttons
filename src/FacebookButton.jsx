import React from 'react';

export default class FacebookButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = ({ initialized: false });
  }

  componentDidMount() {
    if (this.state.initialized) {
      return;
    }

    if (typeof FB === 'undefined') {
      const facebookbutton = this.node;
      const facebookscript = document.createElement('script');
      facebookscript.src = '//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.5';
      facebookscript.id = 'facebook-jssdk';
      facebookbutton.parentNode.appendChild(facebookscript);
    } else {
      FB.XFBML.parse(); // eslint-disable-line
    }

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    return (
      <div
        ref={node => this.node = node}
        className="fb-like"
        data-href={this.props.url}
        data-layout={this.props.layout}
        data-action={this.props.action}
        data-show-faces={this.props.showFaces}
        data-share={this.props.share}
      />
    );
  }
}

FacebookButton.propTypes = {
  url: React.PropTypes.string,
  layout: React.PropTypes.string,
  action: React.PropTypes.string,
  showFaces: React.PropTypes.bool,
  share: React.PropTypes.bool,
};

FacebookButton.defaultProps = {
  url: (typeof window !== 'undefined' ? window.url : ''),
  layout: '',
  action: '',
  showFaces: false,
  share: false
};

/*
<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) return;
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.5";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>
 */
