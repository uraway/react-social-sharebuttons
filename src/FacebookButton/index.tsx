import { FC, useEffect } from 'react';

declare let FB: {
  XFBML: {
    parse: () => void;
  };
};

export type FacebookButtonProps = {
  url?: string;
  layout?: 'standard' | 'box_count' | 'button_count' | 'button';
  action?: 'like' | 'recommended';
  showFaces?: boolean;
  share?: boolean;
};

const FacebookButton: FC<FacebookButtonProps> = (props) => {
  useEffect(() => {
    let script: HTMLScriptElement;
    if (typeof FB === 'undefined') {
      script = document.createElement('script');
      script.src = 'https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v2.5';
      script.id = 'facebook-jssdk';
      document.getElementsByTagName('head')[0]?.appendChild(script);
    } else {
      FB.XFBML.parse();
    }

    return () => script?.remove();
  }, [props]);
  const { url, layout, action, showFaces, share, ...others } = props;
  return (
    <div
      className="fb-like"
      data-href={url}
      data-layout={layout}
      data-action={action}
      data-show-faces={showFaces}
      data-share={share}
      {...others}
    />
  );
};

FacebookButton.defaultProps = {
  url: typeof window !== 'undefined' ? window.location.href : '',
  layout: 'standard',
  action: 'like',
  showFaces: false,
  share: false,
};

export default FacebookButton;
