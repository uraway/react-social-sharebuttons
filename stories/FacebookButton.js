import React from 'react';
import { text, select, boolean } from '@storybook/addon-knobs';
import { FacebookButton } from '../dist/index.esm.js';

const layoutOptions = {
  standard: 'standard',
  box_count: 'box_count',
  button_count: 'button_count',
  button: 'button',
};

export default {
  title: 'FacebookButton',
};

export const Basic = () => (
  <FacebookButton
    url={text('url', '')}
    layout={select('layout', layoutOptions, 'standard')}
    showFaces={boolean('showFaces', false)}
    share={boolean('share', false)}
  />
);
