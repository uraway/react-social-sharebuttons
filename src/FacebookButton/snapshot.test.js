import React from 'react';
import renderer from 'react-test-renderer';
import FacebookButton from '.';

it('renders correctly', () => {
  const tree = renderer.create(<FacebookButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with props', () => {
  const tree = renderer.create(<FacebookButton url="url" layout="standard" showFaces={false} share />).toJSON();
  expect(tree).toMatchSnapshot();
});
