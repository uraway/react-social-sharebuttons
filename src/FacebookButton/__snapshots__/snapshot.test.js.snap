// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="fb-like"
    data-action="like"
    data-layout="standard"
    data-share={false}
    data-show-faces={false}
  />
</div>
`;

exports[`renders correctly with props 1`] = `
<div>
  <div
    className="fb-like"
    data-action="like"
    data-href="url"
    data-layout="standard"
    data-share={true}
    data-show-faces={false}
  />
</div>
`;
